### PR TITLE
feat: 'pkgdb manifest {update,registry}' subcommands

### PR DESCRIPTION
**Pull request by aakropotkin**
- fixes: fix EnvironmentMixin::getLockfilePath and simplify EnvironmentMixinGA inits
- fixes: `Environment::getCombinedRegistry()` to include lockfile inputs.
- fixes: A bug which ignored the `options.systems` field in _manifest_ files in combination with `--ga-registry`.
- adds: `pkgdb manifest registry` sub-command
- adds: equality and ostream ops for lockfile inputs and packages
- adds: `pkgdb manifest update` sub-command

## TODO ( post-GA ):
- `pkgdb manifest lock` should have special handling for inputs pulled from `global-manifest` which are not in `manifest.toml` if they are used to resolve packages ( warn the user ).

### CODE REVIEW

**Review by mkenigs — CHANGES_REQUESTED** [review]
I don't see any tests for `update` or `registry` - could you add something even if it's not in depth?

**Comment by aakropotkin** [comment]
> I don't see any tests for `update` or `registry` - could you add something even if it's not in depth?

Yeah I can add some

File: src/resolver/lockfile.cc [hunk 1]
```diff
@@ -234,6 +234,15 @@ to_json( nlohmann::json & jto, const LockedInputRaw & raw )
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+std::ostream &
+operator<<( std::ostream & oss, const LockedInputRaw & raw )
+{
+  return oss << nlohmann::json( raw ).dump();
```
Comment (mkenigs) [inline src/resolver/lockfile.cc:242]: Nonblocking
```suggestion
  return oss << nlohmann::json( raw ).dump(4);
```

File: src/resolver/lockfile.cc [hunk 2]
```diff
@@ -302,6 +311,15 @@ to_json( nlohmann::json & jto, const LockedPackageRaw & raw )
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+std::ostream &
+operator<<( std::ostream & oss, const LockedPackageRaw & raw )
+{
+  return oss << nlohmann::json( raw ).dump();
```
Comment (mkenigs) [inline src/resolver/lockfile.cc:319]: Nonblocking
```suggestion
  return oss << nlohmann::json( raw ).dump(4);
```

File: src/resolver/lockfile.cc (hunk 1 above)
Comment (aakropotkin) [reply src/resolver/lockfile.cc:244]: What does the 4 do?

Comment (aakropotkin) [reply src/resolver/lockfile.cc:244]: I added the `<<` op for `EXPECT_EQ`.
The test suite is going to format those errors like this if we add newlines and indents:
```
Expectation failed: ( ( input1 ) == ( input2 ) ). Got '{
  "foo": 1,
  "bar": 2
}' != '{
  "quux": 3
}'
```

If you prefer that I'm happy to apply the change; just wanted to highlight the behavior.

**Review by mkenigs — APPROVED** [review]

File: src/resolver/lockfile.cc (hunk 1 above)
Comment (mkenigs) [reply src/resolver/lockfile.cc:244]: I was thinking for some of the large JSON blobs having it formatted would be nicer, e.g. an entire lockfile. I agree it's worse for the first and last line. I don't have super strong feelings

**Pull request merged**